src/ex2/app.js

```diff
@@ -0,0 +1,85 @@
+import ItemManager from "./ItemManager.js";
+import Task from "./Tasks.js";
+import PokemonClient from "./PokemonClient.js";
+import Pokemon from "./Pokemons.js";
+import RenderTaskList from "./RenderList.js";
+
+export default class App {
+  constructor() {
+    this.itemManager = new ItemManager();
+    this.RenderTaskList = new RenderTaskList(this.itemManager,this.itemManager.tasksArray());
+    this.pokemonFetcher = new PokemonClient();
+    this.inputFiled = document.querySelector(".new-input");
+    this.addTaskBtn = document.querySelector(".new-button");
+    this.clearAllTasksBtn = document.querySelector(".clear-all-btn");
+  }
+
+  runApp() {
+    this.RenderTaskList.render(this.itemManager.tasksArray());
+
+    this.addTaskBtn.addEventListener("click", async () => {
+      let inputValue = this.inputFiled.value;
+
+      if (inputValue.trim() === "") {
+        alert("Invalid Optins");
+      } else {
+        if (this.inputOfNums(inputValue)) {
+          const parsedInput = this.checkUserInput(inputValue);
+          try {
+            const pokemons = await this.pokemonFetcher.pokemonFetcherById(parsedInput);
+            this.insertPokemon(pokemons);
+            this.inputFiled.value = "";
+            this.RenderTaskList.render(this.itemManager.tasksArray());
+          } catch (e) {}
+        } else {
+          this.itemManager.addItem(
+            new Task(inputValue, false)
+          );
+          this.inputFiled.value = "";
+          this.RenderTaskList.render(this.itemManager.tasksArray());
+        }
```

Comment:
extract this code to external function, would be reuseable and much easier to maintain